src/Compilers/CSharp/Portable/Binder/LocalBinderFactory.cs

```diff
@@ -233,7 +233,7 @@ public override void VisitLocalFunctionStatement(LocalFunctionStatementSyntax no
 
                     binder = binder.WithUnsafeRegionIfNecessary(node.Modifiers);
 
-                    Visit(body, new InMethodBinder(match, binder));
+                    Visit(body, new InMethodBinder(match, binder, binder.Flags & ~BinderFlags.AllClearedAtExecutableCodeBoundary));
```

Comment:
>AllClearedAtExecutableCodeBoundary [](start = 94, length = 34)

Please make sure to add tests covering each bit from BinderFlags.AllClearedAtExecutableCodeBoundary

Reply:
Like Vlad pointed out, InCatchFilter probably can't be tested, since we can't put a local function inside an expression. It wouldn't change much, but I could define a new group of flags without the InCatchFilter flag.

Reply:
I don't think that's necessary, we can just skip InCatchFilter.